src/applications/appeals/10182/validations/date.js

```diff
@@ -17,26 +17,54 @@ export const validateDate = (errors, rawString = '') => {
   const dateString = fixDateFormat(rawString);
   const { day, month, year } = parseISODate(dateString);
   const date = moment(rawString, FORMAT_YMD);
+  // get last day of the month (month is zero based, so we're +1 month, day 0);
+  // new Date() will recalculate and go back to last day of the previous month
+  const maxDays = year && month ? new Date(year, month, 0).getDate() : 31;
+  const invalidDate = dateString?.length < FORMAT_YMD.length || !date.isValid();
+  const errorParts = {
+    month: !month || month < 1 || month > 12,
+    day: !day || day < 1 || day > maxDays,
+    year: !year,
+    other: false, // catch all for partial & invalid dates
+  };
+
   if (
     !year ||
-    year === '' ||
     !day ||
     day === '0' ||
     !month ||
     month === '0' ||
     dateString?.length < FORMAT_YMD.length
   ) {
-    // errors.addError(issueErrorMessages.missingDecisionDate);
     // The va-date component currently overrides the error message when the
     // value is blank
+    errors.addError(issueErrorMessages.missingDecisionDate);
+    errorParts.other = true; // other part error
+  } else if (
+    errorParts.month ||
+    errorParts.day ||
+    errorParts.year ||
+    invalidDate
+  ) {
     errors.addError(issueErrorMessages.invalidDate);
-  } else if (!date.isValid()) {
-    errors.addError(issueErrorMessages.invalidDate);
+    errorParts.other = true; // other part error
   } else if (date.isSameOrAfter(maxDate)) {
     // Lighthouse won't accept same day (as submission) decision date
     errors.addError(issueErrorMessages.pastDate);
+    errorParts.year = true; // only the year is invalid at this point
```

Comment:
If the logic gets to here, we have valid month, day & year values, but because the year is not in range, we're only setting the year as invalid. The focus will still be moved to the first (month) input though.

Reply:
Got it! Thanks